telemetry/metrics.go

```diff
@@ -0,0 +1,130 @@
+package telemetry
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/armon/go-metrics"
+	metricsprom "github.com/armon/go-metrics/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/expfmt"
+)
+
+// Config defines the configuration options for application telemetry.
+type Config struct {
+	// Prefixed with keys to separate services
+	ServiceName string `mapstructure:"service-name"`
+
+	// Enabled enables the application telemetry functionality. When enabled,
+	// an in-memory sink is also enabled by default. Operators may also enabled
+	// other sinks such as Prometheus.
+	Enabled bool `mapstructure:"enabled"`
+
+	// Enable prefixing gauge values with hostname
+	EnableHostname bool `mapstructure:"enable-hostname"`
+
+	// Enable adding hostname to labels
+	EnableHostnameLabel bool `mapstructure:"enable-hostname-label"`
+
+	// Enable adding service to labels
+	EnableServiceLabel bool `mapstructure:"enable-service-label"`
+
+	// PrometheusRetentionTime, when positive, enables a Prometheus metrics sink.
+	PrometheusRetentionTime int64 `mapstructure:"prometheus-retention-time"`
```

Comment:
Is this seconds?

Reply:
Yes

Reply:
Alright (and excuse my lack of prometheus knowledge), is a negative number an acceptable value? If it is not, maybe we should use `uint64` here.

Reply:
Only >= 0 is accepted. See `New()`.